src/plugins/es_ui_shared/docs/docs/form_lib/core/in_out_raw_state.md

```diff
@@ -0,0 +1,81 @@
+---
+id: in_out_raw_state
+title: In, Out & Raw data states
+sidebar_label: In, Out & Raw data states
+---
+
+It is important to understand the different state of the form data as this will help understanding other concepts likes the `serializers` and `deserializers`.
+
+## "Raw" data state
+
+As you have probably noticed when declaring a field with the `<UseField />` component we don't give it a "_name_" but a "_path_". This gives us a lot of flexibility to declare the final shape of the outputted data of our form, as a path can be any valid [`lodash.set()` path](https://lodash.com/docs/#set).
+
+For example
+
+```js
+
+// Given the following interface for the form
+interface MyForm {
+  user: {
+    name: string;
+    lastName: string;
+  }
+}
+
+// You would declare the following field paths
+<UseField path="user.name" />
+<UseField path="user.lastName" />
+
+// And this would be the raw form data
+const rawFormData = {
+  'user.name': 'John',
+  'user.lastname': 'Snow'
+};
+```
+
+Now try changing the paths to `user[0]` and `user[1]`, can you guess what the output will be?
+
+Being able to declare fields paths improves performance as the form data is saved as a **flat object**, meaning that updating its state is always an O(1) operation. Only when we submit the form, or when we explicityly ask for it, the final object is built and returned to the consumer.
```

Comment:
Similar revision required here after https://github.com/elastic/kibana/issues/78589 is implemented